src/graphforge/executor/evaluator.py

```
@@ -1149,6 +1169,15 @@ def _evaluate_function(
             return CypherInt(len(arg.value))
         raise TypeError(f"SIZE expects list or string, got {type(arg).__name__}")
 
+    # ISEMPTY function for lists, strings, and maps
+    if func_name == "ISEMPTY":
+        arg = args[0]
+        if isinstance(arg, (CypherList, CypherString)):
+            return CypherBool(len(arg.value) == 0)
+        if isinstance(arg, CypherMap):
+            return CypherBool(len(arg.value) == 0)
+        raise TypeError(f"ISEMPTY expects list, string, or map, got {type(arg).__name__}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)